universal-login-sdk/lib/core/observers/PriceObserver.ts

```diff
@@ -1,17 +1,16 @@
 const cryptocompare = require('cryptocompare');
-import {TokenDetails, ObservedCurrency, TokensPrices} from '@universal-login/commons';
+import {ObservedCurrency, TokensPrices} from '@universal-login/commons';
 import ObserverRunner from './ObserverRunner';
 import {PRICE_OBSERVER_DEAFULT_TICK} from '../../config/observers';
+import {TokensDetailsStore} from '../../integration/ethereum/TokensDetailsStore';
 
 export class PriceObserver extends ObserverRunner {
-  private observedTokensSymbols: string[] = [];
   private lastTokenPrices: TokensPrices = {};
   private callbacks: Function[] = [];
 
-  constructor(private observedTokens: TokenDetails[], private observedCurrencies: ObservedCurrency[], step: number = PRICE_OBSERVER_DEAFULT_TICK) {
+  constructor(private tokensDetailsStore: TokensDetailsStore, private observedCurrencies: ObservedCurrency[], step: number = PRICE_OBSERVER_DEAFULT_TICK) {
     super();
     this.step = step;
-    this.observedTokensSymbols = this.observedTokens.map((token) => token.symbol);
```

Comment:
Why?

Reply:
1) Price observer was created(observedTokens) based on config with obj that has {symbol, name, address}. Now config has only address and we can't do it
2) Fetching this information is asynchronous, so we cannot give for PriceObserver which is created in the constructor tokenDetails which doesn't have symbols yet.